### DISCUSSION TITLE
How to write a custom "Client Awareness" plugin

### DISCUSSION THREAD

**Discussion by edude03** [apollographql/router#2039] [Q&A]
Most clients of our super graph use the `user agent` header instead of the apollographql ones. I'd like to track these clients, but presumably, I'll need to parse the user agent into client name and version. Is there any prior art or guidance on how to do this?

**Comment by bnjjj** [comment]
What do you want to do with these `user agent` once parsed ?

**Reply by edude03** [reply]
We use apollo studio, so we want apollo studio to know what client made the request. I'm assuming the solution would look something like

```
let (client_name, client_version) = parse_useragent(headers.get(user_agent))?; //parse user agent being the function I'd write
context.set_client_name(client_name);
context.set_client_version(client_version)
```

**Reply by bnjjj** [reply]
Hmm for now you can't send some of your specific data to apollo studio using a plugin. Was it something you did with the gateway ?
